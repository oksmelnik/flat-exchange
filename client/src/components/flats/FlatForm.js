import React, { Component } from 'react'
import '../Profile.css'

class FlatForm extends Component {
  state = {}

  handleSubmit = (e) => {
    this.props.onSubmit(this.state)
    }

    handleChange = (event) => {
      const {name, value} = event.target

      this.setState({
        [name]: value
        })
      }

  render() {
    return (
      <div className='panel-body'>
        <div className='container'>
          <div className='row'>

            <div className='col-md-4 select'>
              <form className="form-group">
                <label>Home Type</label>
                  <select className="form-control" name='home_type' onChange={this.handleChange}>
                    <option value="" selected>Choose here</option>
                    <option value="Appartment">Appartment</option>
                    <option value="House">House</option>
                    <option value="Bed&Breakfast">Bed&Breakfast</option>
                    </select>
                  </form>
                </div>

              <div className='col-md-4 select'>
                <div className="form-group">
                  <label>Room Type</label>
                    <select className="form-control" name='room_type' onChange={this.handleChange}>
                      <option value="" selected>Choose here</option>
                      <option value="Entire">Entire</option>
                      <option value="Shared">Shared</option>
                      </select>
                    </div>
                  </div>

              <div className='col-md-4 select'>
                <div className="form-group">
                  <label>Accomodate</label>
                      <select className="form-control" name='accomodate' onChange={this.handleChange}>
                        <option value="" selected>Choose here</option>
                        <option value="1">1</option>
                        <option value="2">2</option>
                        <option value="3">3</option>
                        <option value="4">4</option>
                        <option value="5">5</option>
                        </select>
                      </div>
                    </div>
                    <div class="text-center">

                      <button
                        type="submit"
                        className="btn btn-normal"
                        onClick={this.handleSubmit}>
                        Submit
                      </button>
                    </div>
                </div>
            </div>
          </div>
        )}}

  export default FlatForm;
