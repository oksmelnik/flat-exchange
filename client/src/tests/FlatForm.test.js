import React from "react";
import { shallow } from "enzyme";
import FlatForm from "../components/flats/FlatForm";
import renderer from "react-test-renderer";

describe("<FlatForm component />", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<FlatForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("have 3 fields", () => {
    const wrapper = shallow(<FlatForm />);
    const count = wrapper.find(".form-group");
    expect(count.length).toEqual(3);
  });

  it("starts with an empty array", () => {
    const wrapper = shallow(<FlatForm />);
    const text = wrapper
      .find("option [selected]")
      .at(0)
      .text();
    expect(text).toEqual("Choose here");
  });

  it("starts with an empty array", () => {
    const wrapper = shallow(<FlatForm />);
    const text = wrapper
      .find("option [selected]")
      .at(1)
      .text();
    expect(text).toEqual("Choose here");
  });

  it("starts with an empty array", () => {
    const wrapper = shallow(<FlatForm />);
    const text = wrapper
      .find("option [selected]")
      .at(2)
      .text();
    expect(text).toEqual("Choose here");
  });
});
