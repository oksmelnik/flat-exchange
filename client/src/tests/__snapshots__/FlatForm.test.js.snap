// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlatForm component /> matches the snapshot 1`] = `
<div
  className="panel-body"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4 select"
      >
        <form
          className="form-group"
        >
          <label>
            Home Type
          </label>
          <select
            className="form-control"
            name="home_type"
            onChange={[Function]}
          >
            <option
              id="home_type"
              selected={true}
            >
              Choose here
            </option>
            <option
              value="Appartment"
            >
              Appartment
            </option>
            <option
              value="House"
            >
              House
            </option>
            <option
              value="Bed&Breakfast"
            >
              Bed&Breakfast
            </option>
          </select>
        </form>
      </div>
      <div
        className="col-md-4 select"
      >
        <div
          className="form-group"
        >
          <label>
            Room Type
          </label>
          <select
            className="form-control"
            name="room_type"
            onChange={[Function]}
          >
            <option
              selected={true}
              value=""
            >
              Choose here
            </option>
            <option
              value="Entire"
            >
              Entire
            </option>
            <option
              value="Shared"
            >
              Shared
            </option>
          </select>
        </div>
      </div>
      <div
        className="col-md-4 select"
      >
        <div
          className="form-group"
        >
          <label>
            Accomodate
          </label>
          <select
            className="form-control"
            name="accomodate"
            onChange={[Function]}
          >
            <option
              selected={true}
              value=""
            >
              Choose here
            </option>
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
          </select>
        </div>
      </div>
      <div
        class="text-center"
      >
        <button
          className="btn btn-normal"
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </div>
</div>
`;
